nucypher/cli/main.py

```diff
@@ -452,150 +470,140 @@ def ursula(click_config,
         ursula_config.miner_agent.confirm_activity(node_address=checksum_address)
+        return
 
-    elif action == 'divide':
+    elif action == 'divide-stake':
         """Divide an existing stake by specifying the new target value and end period"""
 
-        stakes = ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address)
+        stakes = list(ursula_config.miner_agent.get_all_stakes(miner_address=checksum_address))
         if len(stakes) == 0:
             raise RuntimeError("There are no active stakes for {}".format(checksum_address))
 
-        if not index:
+        if index is None:
             for selection_index, stake_info in enumerate(stakes):
                 click.echo("{} ....... {}".format(selection_index, stake_info))
             index = click.prompt("Select a stake to divide", type=click.INT)
 
-        target_value = click.prompt("Enter new target value", type=click.INT)
-        extension = click.prompt("Enter number of periods to extend", type=click.INT)
+        if not value:
+            target_value = click.prompt("Enter new target value", type=click.INT)
+        else:
+            target_value = value
 
-        click.echo("""
-        Current Stake: {}
+        if not duration:
+            extension = click.prompt("Enter number of periods to extend", type=click.INT)
+        else:
+            extension = duration
 
-        New target value {}
-        New end period: {}
+        if not force:
+            click.echo("""
+            Current Stake: {}
+    
+            New target value {}
+            New end period: {}
+    
+            """.format(stakes[index],
+                       target_value,
+                       target_value + extension))
 
-        """.format(stakes[index],
-                   target_value,
-                   target_value + extension))
+            click.confirm("Is this correct?", abort=True)
 
-        click.confirm("Is this correct?", abort=True)
-        ursula_config.miner_agent.divide_stake(miner_address=checksum_address,
-                                               stake_index=index,
-                                               value=value,
-                                               periods=extension)
+        miner = Miner(is_me=True,
+                      checksum_address=ursula_config.checksum_public_address,
+                      blockchain=ursula_config.blockchain)
 
-    elif action == 'collect-reward':          # TODO: Implement
+        txhash_bytes = miner.divide_stake(stake_index=index,
+                                          target_value=value,
+                                          additional_periods=duration)
+
+        if not quiet:
+            click.secho('Successfully divided stake', fg='green')
+            click.secho(f'Transaction Hash ........... {txhash_bytes.hex()}')
+
+        return
+
+    elif action == 'collect-reward':
```

Comment:
It seems to me that this structure of multiple commands which of them has multiple `@option`s, with if/else on top of that, all in one file, is sort of hard to read.
Perhaps, what would make it more readable is splitting the cli into multiple commands (like git does).
But that's probably not something to worry about in this PR

Reply:
To be honest I think the logical division is quite understandable and sound:  Each character has a CLI entry point, `nucypher ursula` for example. Then contract deployment and preallocation holders also have entry points: `nucypher-prealloc` and `nucypher-deploy`.

Ultimately:

```
nucypher ursula
nucypher alice
nucypher bob
nucypher-prealloc
nucypher-deploy
```

Are people not liking this command structure?